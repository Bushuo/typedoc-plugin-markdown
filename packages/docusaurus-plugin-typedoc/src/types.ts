export interface PluginOptions {
  id: string;
  docsRoot: string;
  out: string;
  sidebar: SidebarOptions | null;
  readmeTitle: string | undefined;
  globalsTitle?: string | undefined;
  plugin?: string[];
  readme?: string;
  disableOutputCheck?: boolean;
  entryPoints?: any;
  entryDocument: string;
  hideInPageTOC: boolean;
  hideBreadcrumbs: boolean;
  siteDir?: string;
}

export interface FrontMatter {
  id: string;
  title: string;
  slug?: string;
  sidebar_label?: string;
  hide_title?: boolean;
}

export interface SidebarOptions {
  fullNames: boolean;
  sidebarFile: string;
  indexLabel?: string;
  readmeLabel: string;
}

export interface Sidebar {
  [sidebarId: string]: SidebarItem[];
}

export interface SidebarCategory {
  type: string;
  label: string;
  items: SidebarItem[];
}

export type SidebarItem = SidebarCategory | string;
