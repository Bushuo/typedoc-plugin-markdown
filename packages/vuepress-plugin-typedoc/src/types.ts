import { Application, ProjectReflection } from 'typedoc';

export interface PluginOptions {
  hideBreadcrumbs: boolean;
  out: string;
  sidebar?: SidebarOptions | null;
  plugin: string[];
  watch: boolean;
}

export interface SidebarOptions {
  fullNames: boolean;
  parentCategory: string;
  sidebarFile: string;
  sidebarPath: string;
}

export interface LoadedContent {
  app: Application;
  project: ProjectReflection;
}
