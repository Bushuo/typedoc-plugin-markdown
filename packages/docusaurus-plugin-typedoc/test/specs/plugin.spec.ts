import { LoadContext } from '@docusaurus/types';
import * as fs from 'fs-extra';
import * as tmp from 'tmp';

import typedocPlugin from '../../dist/plugin';

tmp.setGracefulCleanup();

async function bootstrap(tmpobj: tmp.DirResult, customOptions = {}) {
  const options = {
    id: 'default',
    logger: 'none',
    entryPoints: ['../typedoc-plugin-markdown/test/stubs/src/theme.ts'],
    tsconfig: '../typedoc-plugin-markdown/test/stubs/tsconfig.json',
  } as any;

  const plugin = typedocPlugin(
    {
      siteDir: tmpobj.name,
      generatedFilesDir: '',
      siteConfig: {},
    } as LoadContext,
    {
      ...options,
      ...customOptions,
    },
  );
  return await plugin;
}

describe(`Plugin:`, () => {
  describe(`(docs)`, () => {
    let tmpobj;
    beforeAll(async () => {
      tmpobj = tmp.dirSync();
      await bootstrap(tmpobj);
    });
    test(`should render`, () => {
      const files = fs.readdirSync(tmpobj.name + '/docs/api');
      expect(files).toMatchSnapshot();
    });

    test(`should write doc`, () => {
      const sidebar = fs.readFileSync(tmpobj.name + '/docs/api/index.md');
      expect(sidebar.toString()).toMatchSnapshot();
    });

    test(`should write category yaml`, () => {
      const sidebar = fs.readFileSync(
        tmpobj.name + '/docs/api/classes/_category_.yml',
      );
      expect(sidebar.toString()).toMatchSnapshot();
    });
  });
});
