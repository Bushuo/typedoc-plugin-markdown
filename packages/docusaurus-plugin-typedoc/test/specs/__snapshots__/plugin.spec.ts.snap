// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin: (docs) should render 1`] = `
Array [
  "classes",
  "enums",
  "index.md",
  "interfaces",
  "modules",
  "modules.md",
]
`;

exports[`Plugin: (docs) should write doc 1`] = `
"---
id: \\"index\\"
title: \\"typedoc-plugin-markdown\\"
slug: \\"/api\\"
sidebar_label: \\"Readme\\"
custom_edit_url: null
---

# MY README

> Some block quote

<!-- some comments -->

Some text
"
`;

exports[`Plugin: (docs) should write sidebar 1`] = `
"module.exports = [
  \\"api/index\\",
  \\"api/modules\\",
  {
    \\"type\\": \\"category\\",
    \\"label\\": \\"Namespaces\\",
    \\"items\\": [
      \\"api/modules/namespacea\\",
      \\"api/modules/namespaceb\\"
    ]
  },
  {
    \\"type\\": \\"category\\",
    \\"label\\": \\"Enumerations\\",
    \\"items\\": [
      \\"api/enums/enumitema\\",
      \\"api/enums/enumitemb\\"
    ]
  },
  {
    \\"type\\": \\"category\\",
    \\"label\\": \\"Classes\\",
    \\"items\\": [
      \\"api/classes/namespacea.somenestedclass\\",
      \\"api/classes/classitema\\",
      \\"api/classes/classitemb\\"
    ]
  },
  {
    \\"type\\": \\"category\\",
    \\"label\\": \\"Interfaces\\",
    \\"items\\": [
      \\"api/interfaces/interfaceitema\\",
      \\"api/interfaces/interfaceitemb\\"
    ]
  }
];"
`;
